src/libraries/System.Private.CoreLib/src/System/Environment.Linux.cs

```diff
@@ -2,11 +2,35 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
 
 namespace System
 {
     public static partial class Environment
     {
         public static long WorkingSet => (long)(Interop.procfs.TryReadStatusFile(Interop.procfs.ProcPid.Self, out Interop.procfs.ParsedStatus status) ? status.VmRSS : 0);
+
+        /// <summary>
+        /// Get the CPU usage, including the process time spent running the application code, the process time spent running the operating system code,
+        /// and the total time spent running both the application and operating system code.
+        /// </summary>
+        [SupportedOSPlatform("maccatalyst")]
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
+        public static ProcessCpuUsage CpuUsage
+        {
+            get
+            {
+                Interop.procfs.ParsedStat stat = GetStat();
+                return new ProcessCpuUsage { UserTime = TicksToTimeSpan(GetStat().Utime), PrivilegedTime = TicksToTimeSpan(GetStat().Stime) };
+            }
+        }
+
+        private static Interop.procfs.ParsedStat GetStat()
```

Comment:
We use procfs in Process because it needs to be able to get these values for any process. But here we're only concerned about the current process. Can we instead use getrusage? It should be way more efficient and also simpler. We can then also change Process to special-case the current process to call to Environment.CpuUsage.

Reply:
Done. Thanks!